### PR TITLE
Improve Slack citations rendering

### PR DESCRIPTION
**Pull request by flvndvd**
## Description

This PR enhances the formatting of citations in Slack messages. Currently, citations can be cluttered and unclear. This update ensures that citations will consistently appear as a type of footer note, with additional descriptions. We will display the name up to 20 characters.

**Before:**
image

**After:**

image

### CODE REVIEW

File: types/src/shared/utils/string_utils.ts [hunk 1]
```diff
@@ -69,3 +69,7 @@ export function redactString(str: string, n: number) {
   const redacted = "•".repeat(str.length - n) + str.slice(-n);
   return redacted;
 }
+
+export function truncate(text: string, length: number, omission = "...") {
```
Comment (spolu) [inline types/src/shared/utils/string_utils.ts:73]: We must already have this somewhere, should try to dry it? I think we have one in connectors, now that it's here it should be shared?

**Review by spolu — APPROVED** [review]
Looks reasonable. `truncate` likely already exits elsewhere

File: types/src/shared/utils/string_utils.ts (hunk 1 above)
Comment (flvndvd) [reply types/src/shared/utils/string_utils.ts:73]: It does not exist, everyone went with there own `.substring()` implementation 🙃.

File: connectors/src/connectors/slack/chat/blocks.ts [hunk 2]
```diff
@@ -38,24 +42,57 @@ function makeMarkdownBlock(text: string) {
...
-    ? [makeDividerBlock(), makeConversationLinkContextBlock(conversationUrl)]
-    : [];
+  const { isThinking, text, footnotes } = messageUpdate;
 
   return {
     blocks: [
       isThinking ? makeThinkingBlock() : makeMarkdownBlock(text),
-      // Bundle the conversation url in the context.
-      ...conversationUrlBlocks,
+      ...makeContextSectionBlocks(conversationUrl, footnotes),
```
Comment (JulesBelveze) [inline connectors/src/connectors/slack/chat/blocks.ts:95]: Do we want a `DividerBlock` even if we end up without footnotes?

File: connectors/src/connectors/slack/chat/citations.ts [hunk 3]
```diff
@@ -18,39 +29,62 @@ export function annotateCitations(
...
 
-          return "";
-        })
-        .join(" ");
-    });
+                return `[${refCounter[k]}]`;
+              }
+
+              if (ref) {
+                if (!refCounter[k]) {
```
Comment (JulesBelveze) [inline connectors/src/connectors/slack/chat/citations.ts:67]: Those conditions are already met right? (line 44-45)

**Pull request merged**